torch/_functorch/aot_autograd.py

```diff
@@ -1352,10 +1352,13 @@ def _aot_export_function(
     flat_args, in_spec = pytree.tree_flatten((args, kwargs))
 
     dynamic_shapes = False
-    for x in flat_args:
-        if isinstance(x, FakeTensor):
-            dynamic_shapes = x.fake_mode.shape_env is not None
-            break
+    if len(flat_args) == 0:
+        dynamic_shapes = FakeTensorMode()
```

Comment:
dynamic_shapes is a bool?

Reply:
oh right. I changed it to True now. Is this the right approach?

Reply:
mm no I don't think you can just say if there are no arguments then it's definitely dynamic..

Reply:
Do you have any pointers/ high level suggestion on how we can find out if it's dynamic or now?